### DISCUSSION TITLE
Separate KTON Staking from Collator Staking while keeping incentive for ring pool and kton pool unchanged.

### DISCUSSION THREAD

**Discussion by hackfisher** [darwinia-network/darwinia#1393] [General]
Generally, a system can have multiple Staking mechanisms coexisting, each with its own incentive purposes. However, Darwinia's current Collator hybrid Staking system has introduced many problems. By mixing different incentive goals together, the system becomes difficult to understand. It also forces these incentives to be bound together, leading to constrained actions, which is a poor design.

The current Collator Staking consists of the ring pool and kton pool, each sharing 50% of the power. While they help in selecting the Collator List, they also share a total of 40% of the new issuance RING incentives, which means 20% each from the new issuance RING.

It is suggested to separate the KTON pool from the Collator Staking, creating an independent KTON Staking module that directly shares 20% of the new issuance RING incentives. The original Collator Staking would then be entirely made up of the RING pool, helping to select the collator list while also sharing the same 20% of the new issuance RING incentives as before. With these changes, KTON Staking will gain more visibility than before, enhancing the effectiveness of incentives. Collators will no longer share commissions from the KTON pool, but considering their role in the system is primarily to maintain liveness, the commission incentives from the RING pool should be entirely sufficient. Additionally, the confusing and hard-to-communicate concept of Power in the current Collator Staking can be dismissed. Clarity in naming and concepts is crucial and will aid in community discussions and in clarifying viewpoints. It’s also worth re-emphasizing that the economic incentives for both the ring pool and kton pool are intended to remain the same as they currently are, to avoid discussions related to the amount of incentives.

Note: Engineers should be reminded to pay attention to the code related to account migration from 1.0 to 2.0, ensuring proper handling and awareness of the above changes.

**Comment by SasoLithops** [comment]
Very interesting suggestion,

I have a few questions how this would work and if I understand it well.

How will new KTON be minted? Will original collator staking remain the same? With 1-36 months locking and this will still influence the ammount of minted KTON but have no influence on getting more Ring?

And if I understand correctly independant Kton staking will share its Ring rewards in way already discussed economic incentives excluding collators share.

My only doubt is if it will be worth for collators and if their commisions will cover the costs and really be entierly sufficient. But it is up to them to share their opinion here. 

Other then that this is good idea to remove POWER and this will be really rewarding for KTON holders.

**Reply by hackfisher** [reply]
> How will new KTON be minted?

The same as currently, from deposit module. 

In future, if there is such a KTON DAO, and one day would like to reclaim the monetary policy rights for minting KTON and cease providing prepaid interest for depositing RING. In exchange, the governance system might adjust the incentive policy for KTON accordingly. However, there is no need to make changes now, and it might be a long time before this topic is discussed. Nonetheless, establishing a KTON DAO and developing other utilities for KTON could be a good idea.

> Will original collator staking remain the same?

Will consist solely of the ring pool, technically not the same, but incentive for ring pool and kton pool unchanged.

> With 1-36 months locking and this will still influence the amount of minted KTON but have no influence on getting more Ring?

No influence on both.

> And if I understand correctly independant Kton staking will share its Ring rewards in way already discussed economic incentives excluding collators share.

Yes, the collator commission is generally small, but it does exist, so KTON Staking yields a bit more than before.

> My only doubt is if it will be worth for collators and if their commisions will cover the costs and really be entierly sufficient. But it is up to them to share their opinion here.

Market competition exists, so there's no need to worry too much, as they can adjust their commission.

**Reply by SasoLithops** [reply]
Thank you for clearing this up.

**Comment by sonoferin2000** [comment]
depositing ring into a term lock,to get kton isnt staking, its just agreeing to lock it down for a term period.so this recommendation he is making shouldn't have any impact on the creation of additional KTON. I like this suggestion. would be interested to hear other viewpoints on this.

**Reply by SasoLithops** [reply]
Yes you are right. It should not. Great to see you join in more debates👍

**Comment by 2xJDubs** [comment]
I really like this change as I have found over the years that Power can be confusing to new users.  Doing this will bring more clarity to staking.  Would it be possible to add to the staking dashboard the total number of RING staked and total number of KTON staked?  This would help users understand their expected rewards and returns using a spreadsheet I am working on.  With respect to RING rewards would a delegators ratio be based on total number of RING staked or the number of RING staked on the collator they are delegating to?

**Comment by sonoferin2000** [comment]
not only for new users, ive been staking since mainet lauch,and i still
dont have a firm grasp on it.I dont really think a spreadsheat here would
be much helpful. power seems to be such a fluid thing,a new person coming
in and staking,or someone unbonding, it changes the numbers. and theres no
counting blocks befor they hit them.

**Comment by hackfisher** [comment]
Switching from the old Collator KTON Staking to the new KTON Staking will undergo a 30-day transition period. During this period, the Power of KTON will linearly decrease to zero. The incentives for the old KTON Pool in Collator Staking will also be reduced accordingly. The reduced portion will be transferred to the new KTON Staking until the transition period is complete. The old Collator Staking will then undergo a Runtime Migration to return the KTON in the old KTON Pool to the users, completing DIP-5. https://github.com/darwinia-network/DIPs/pull/15


~~This represents the proportion of the issuance that will be transferred to the new KTON Staking over time, ultimately reaching 20%.~~

**Reply by hackfisher** [reply]
The reason for introducing a transition period, rather than a one-off data and asset migration, is to avoid forced transfers as much as possible. Although a one-time data and asset migration may seem more convenient because it does not require on-chain operations and allows users to passively wait, it introduces potential accidents. The first type of accident arises from the need to migrate KTON to the new Staking contract using Runtime for a forced transfer, which violates common procedural rules and may carry the risk of some social contract inconsistencies. Moreover, it could lead to users not knowing where their KTON went before receiving notification, which, to some extent, undermines the user experience.

The approach for the transition period can primarily focus on modifications to the implementation of fiscal policies and the closure of the old KTON Staking.

**Comment by Afolabigram1** [comment]
yeah 0

**Comment by sonoferin2000** [comment]
Could there be a mechanism integrated into staking to enable automatic locking of rewards for a specified time period, with the option for KTON rewards to be automatically added to our stake, eliminating the need for manual intervention each time? It would be convenient to have the choice to put staking on autopilot. The ability to accumulate rewards and trigger automatic locking when they reach a predetermined amount, with KTON rewards seamlessly staked, would save considerable time. Not everyone can regularly check and manage their staking activities. Having the option to periodically check in to adjust delegation if necessary would be beneficial.

**Reply by hackfisher** [reply]
After upgrade, these can be done by introducing extra contract supporting enhanced features, including the scope you mentioned, Liquid staking and governance support etc.

**Comment by wuminzhe** [comment]
![kton staking](https://github.com/darwinia-network/darwinia/assets/1608576/cb849812-126c-40a7-b8f3-66a099c42d3f)
